### DISCUSSION TITLE
MincostFlow Algorithm

### DISCUSSION THREAD

**Discussion by dogactopay** [networkx/networkx#5447] [Q&A] [answered]
Greetings,
We are using Minimum Cost Flow method in our graduation project, Can you please inform me about the working algorithm back?
Thanks in advance :)

**Comment by MridulS** [comment, marked as answer]
min_cost_flow is using a primal network simplex algorithm underneath. You can look at the references in the documentation for more information :)

https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.flow.network_simplex.html

**Comment by rossbar** [comment]
Maybe not particularly relevant to the initial question, but just in case you're interested in flow problems more generally, nx-guides has a very nice tutorial that delves into the Dinitz algorithm (and some applications) in much more detail: 

https://networkx.org/nx-guides/content/algorithms/flow/dinitz_alg.html
